src/Microsoft.SqlTools.ServiceLayer/LanguageServices/LanguageService.cs

```diff
@@ -281,6 +282,39 @@ public void InitializeService(ServiceHost serviceHost, SqlToolsContext context)
 
         #region Request Handlers
 
+        /// <summary>
+        /// T-SQL syntax parse request callback
+        /// </summary>
+        /// <param name="param"></param>
+        /// <param name="requestContext"></param>
+        /// <returns></returns>
+        internal async Task HandleSyntaxParseRequest(SyntaxParseParams param, RequestContext<SyntaxParseResult> requestContext)
+        {            
+            try
+            {            
+                ParseResult result = Parser.Parse(param.Query);
+                SyntaxParseResult syntaxResult = new SyntaxParseResult();
+                if (result != null && result.Errors.Count() == 0)
+                {
+                    syntaxResult.Parseable = true;
+                } else 
+                {
+                    syntaxResult.Parseable = false;
+                    string[] errorMessages = {};
```

Comment:
why not preallocate the array since you know the desired size `Errors.Count()`, which would avoid needing to dynamically grow the array.